File: Codeforces/Contests/Codeforces Round #794 (Div.2)/CircularLocalMinimax.py
t = int(input(''))

for _ in range(t):
    n = int(input(''))
    arr = [int(i) for i in input('').split(' ')]
    solved = []

    if n%2!=0:
        print('NO')

    curr = min(arr)
    arr.remove(curr)
    solved.append(curr)
    ind = 0
    impossible = False
    while ind < n:
        if ind % 2 ==0:
            max_val = max(arr)
            arr.remove(max_val)
            if max_val <= curr:
                impossible = True
                break
            curr = max_val
            ind += 1
        elif ind % 2 != 0:
            min_val = min(arr)
            arr.remove(min_val)
            if min_val >= curr:
                impossible = True
                break
            curr = min_val
            ind+=1

File: Codeforces/Contests/Codeforces Round #794 (Div.2)/CircularLocalMinimax.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {

    int t;
    cin>>t;
    while(t--){
        int n;
        cin>>n;
        vector<int>arr(n);
        vector<int>solved(n);
        for (int i = 0; i < n; i++){
            cin>>arr[i];
        }

        if (n% 2 != 0){
            cout<<"NO"<<endl;
        }

        int curr = *min_element(arr.begin(), arr.end());
        int ind = 0;

        while (ind < n) {

        }

    }

}